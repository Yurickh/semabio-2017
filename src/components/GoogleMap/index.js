import React, { Component } from 'react'
import PropTypes from 'prop-types'
import GoogleMapsLoader from 'google-maps'
import { googleMapsKey } from '../../secret-keys'

import './styles.css'

const colorForest = '#58C166'
const colorWoods = '#3A7742'

class GoogleMap extends Component {
  componentDidMount() {
    const { lat, lng, zoom } = this.props

    GoogleMapsLoader.KEY = googleMapsKey()

    GoogleMapsLoader.load(google => {
      this.mapAPI = new google.maps.Map(this.mapNode, {
        center: { lat, lng },
        zoom: zoom,
        styles: [
          {elementType: 'geometry', stylers: [{color: '#242f3e'}]},
          {elementType: 'labels.text.stroke', stylers: [{color: '#242f3e'}]},
          {elementType: 'labels.text.fill', stylers: [{color: colorWoods}]},
          {
            featureType: 'administrative.locality',
            elementType: 'labels.text.fill',
            stylers: [{color: colorForest}]
          },
          {
            featureType: 'poi',
            elementType: 'labels.text.fill',
            stylers: [{color: colorForest}]
          },
          {
            featureType: 'poi.park',
            elementType: 'geometry',
            stylers: [{color: '#263c3f'}]
          },
          {
            featureType: 'poi.park',
            elementType: 'labels.text.fill',
            stylers: [{color: colorWoods}]
          },
          {
            featureType: 'road',
            elementType: 'geometry',
            stylers: [{color: '#38414e'}]
          },
          {
            featureType: 'road',
            elementType: 'geometry.stroke',
            stylers: [{color: '#212a37'}]
          },
          {
            featureType: 'road',
            elementType: 'labels.text.fill',
            stylers: [{color: colorWoods}]
          },
          {
            featureType: 'road.highway',
            elementType: 'geometry',
            stylers: [{color: '#38414e'}]
          },
          {
            featureType: 'road.highway',
            elementType: 'geometry.stroke',
            stylers: [{color: '#1f2835'}]
          },
          {
            featureType: 'road.highway',
            elementType: 'labels.text.fill',
            stylers: [{color: colorWoods}]
          },
          {
            featureType: 'transit',
            elementType: 'geometry',
            stylers: [{color: '#2f3948'}]
          },
          {
            featureType: 'transit.station',
            elementType: 'labels.text.fill',
            stylers: [{color: colorForest}]
          },
          {
            featureType: 'water',
            elementType: 'geometry',
            stylers: [{color: '#17263c'}]
          },
          {
            featureType: 'water',
            elementType: 'labels.text.fill',
            stylers: [{color: colorWoods}]
          },
          {
            featureType: 'water',
            elementType: 'labels.text.stroke',
            stylers: [{color: '#38414e'}]
          }
        ]
      })

      this.marker = new google.maps.Marker({
        position: { lat, lng },
        map: this.mapAPI,
      })
    })
  }

  static propTypes = {
    lat: PropTypes.number.isRequired,
    lng: PropTypes.number.isRequired,
    zoom: PropTypes.number.isRequired,
  }

  render() {
    return (
      <div
        className="google-map"
        ref={map => this.mapNode = map}
      />
    )
  }
}

export default GoogleMap
