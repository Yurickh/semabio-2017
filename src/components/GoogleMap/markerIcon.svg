<?xml version="1.0" encoding="UTF-8"?>
<svg width="51px" height="65px" viewBox="0 0 51 65" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
    <!-- Generator: Sketch 45.2 (43514) - http://www.bohemiancoding.com/sketch -->
    <title>marker</title>
    <desc>Created with Sketch.</desc>
    <defs>
        <ellipse id="path-1" cx="29.5" cy="46" rx="17.5" ry="13"></ellipse>
        <filter x="-20.0%" y="-19.2%" width="140.0%" height="153.8%" filterUnits="objectBoundingBox" id="filter-2">
            <feOffset dx="0" dy="2" in="SourceAlpha" result="shadowOffsetOuter1"></feOffset>
            <feGaussianBlur stdDeviation="2" in="shadowOffsetOuter1" result="shadowBlurOuter1"></feGaussianBlur>
            <feColorMatrix values="0 0 0 0 0   0 0 0 0 0   0 0 0 0 0  0 0 0 0.5 0" type="matrix" in="shadowBlurOuter1"></feColorMatrix>
        </filter>
    </defs>
    <g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
        <g id="BetterDesk" transform="translate(-334.000000, -5279.000000)">
            <g id="marker" transform="translate(334.000000, 5279.000000)">
                <g id="shadow" fill-opacity="1" fill="black">
                    <use filter="url(#filter-2)" xlink:href="#path-1"></use>
                </g>
                <path d="M19.5,0 C8.73321429,0 0,8.778 0,19.6 C0,34.3 19.5,56 19.5,56 C19.5,56 39,34.3 39,19.6 C39,8.778 30.2667857,0 19.5,0 L19.5,0 Z M19.5,26.6 C15.6557143,26.6 12.5357143,23.464 12.5357143,19.6 C12.5357143,15.736 15.6557143,12.6 19.5,12.6 C23.3442857,12.6 26.4642857,15.736 26.4642857,19.6 C26.4642857,23.464 23.3442857,26.6 19.5,26.6 L19.5,26.6 Z" id="Shape" fill="#58C166"></path>
            </g>
        </g>
    </g>
</svg>