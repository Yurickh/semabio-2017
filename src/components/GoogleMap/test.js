import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import GoogleMap from '../GoogleMap'

describe('GoogleMap', () => {
  it('should render correctly', () => {
    const map = shallow(<GoogleMap lat={0} lng={0} zoom={4} />)
    expect(toJSON(map)).toMatchSnapshot()
  })
})
