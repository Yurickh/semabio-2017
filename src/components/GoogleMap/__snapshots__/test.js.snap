// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleMap should render correctly 1`] = `
<div
  className="google-map"
/>
`;
